text/0000-yarn-berry-support.md

```diff
@@ -0,0 +1,65 @@
+- Feature Name: yarn_berry_support
+- Start Date: 2022-02-16
+- RFC PR: (leave this empty)
+- Volta Issue: https://github.com/volta-cli/volta/issues/651
+
+# Summary
+[summary]: #summary
+
+Roadmap of the necessary changes to support Yarn Berry (Yarn 2.0+)
+
+# Motivation
+[motivation]: #motivation
+
+We currently only support legacy Yarn 1.x installation and pinning. While Yarn 2.x had a lot of infrastructure churn that made it difficult to support, Yarn 3.0+ is more stable and provides all of the pieces that we need to add support. To maintain compatibility as Yarn continues to evolve, we should support Yarn Berry.
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+The ideal outcome with this change is that there will be very little we need to teach users. Newer versions of Yarn will be supported and users will be able to continue to use Volta the way they always have, without needing to worry about specific details of their setup.
+
+The main detail we will need to communicate is the lack of support for the 2.x major version of Yarn (only supporting Yarn 1.x and Yarn 3.0+).
```

Comment:
We will also need to communicate how users should handle an upgrade from Yarn 1.x --> Yarn 3.x with Volta.

The [yarn migration docs](https://yarnpkg.com/getting-started/migration#step-by-step) are a good resource for this, but Volta users should not need to run `npm install -g yarn` or `yarn set version berry` (although we could potentially intercept those commands and Do The Right Thing ™️ )

Reply:
Great call-out, we can definitely document how to use newer Yarn versions with Volta. AFAICT, Yarn Berry doesn't have "global" commands, so really it should be `volta pin yarn@3` and that's all you need.